x/concentrated-liquidity/swapstrategy/one_for_zero.go

```diff
@@ -54,40 +55,59 @@ func (s oneForZeroStrategy) GetSqrtTargetPrice(nextTickSqrtPrice sdk.Dec) sdk.De
 //
 // OneForZero details:
 // - oneForZeroStrategy assumes moving to the right of the current square root price.
-func (s oneForZeroStrategy) ComputeSwapWithinBucketOutGivenIn(sqrtPriceCurrent, sqrtPriceTarget, liquidity, amountOneInRemaining sdk.Dec) (sdk.Dec, sdk.Dec, sdk.Dec, sdk.Dec) {
+func (s oneForZeroStrategy) ComputeSwapWithinBucketOutGivenIn(sqrtPriceCurrent osmomath.BigDec, sqrtPriceTarget, liquidity, amountOneInRemaining sdk.Dec) (osmomath.BigDec, sdk.Dec, sdk.Dec, sdk.Dec) {
+	sqrtPriceTargetBigDec := osmomath.BigDecFromSDKDec(sqrtPriceTarget)
+	liquidityBigDec := osmomath.BigDecFromSDKDec(liquidity)
+	amountOneInRemainingBigDec := osmomath.BigDecFromSDKDec(amountOneInRemaining)
+
 	// Estimate the amount of token one needed until the target sqrt price is reached.
-	amountOneIn := math.CalcAmount1Delta(liquidity, sqrtPriceTarget, sqrtPriceCurrent, true) // N.B.: if this is false, causes infinite loop
+	amountOneIn := math.CalcAmount1Delta(liquidityBigDec, sqrtPriceTargetBigDec, sqrtPriceCurrent, true)
+
+	fmt.Println("liquidity", liquidity)
+	fmt.Println("sqrtPriceCurrent", sqrtPriceCurrent)
+	fmt.Println("sqrtPriceTarget", sqrtPriceTarget)
+
+	fmt.Println("amountOneInRemainingBigDec", amountOneInRemainingBigDec)
+	fmt.Println("amountOneIn start", amountOneIn)
 
 	// Calculate sqrtPriceNext on the amount of token remaining after spread reward.
-	amountOneInRemainingLessSpreadReward := amountOneInRemaining.Mul(sdk.OneDec().Sub(s.spreadFactor))
+	amountOneInRemainingLessSpreadReward := amountOneInRemainingBigDec.MulTruncate(oneBigDec.Sub(osmomath.BigDecFromSDKDec(s.spreadFactor)))
```

Comment:
The primary reason is that we want to avoid overpaying amount out to the user. The value of `amountOut` depends on the `sqrtPriceNext`. `sqrtPriceNext` depends on `amountOneInRemainingLessSpreadReward`. The smaller the `amountOneInRemainingLessSpreadReward`, the closer the `sqrtPriceNext` will be to the `sqrtPriceCurrent`, and the smaller the final `amountOut` consumed will be.